template-parts/post-list-filters.php

```diff
@@ -0,0 +1,143 @@
+ *
+ * @package shiro
+ */
+
+?>
+
+<section class="post-list-filter mw-980">
+
+	<div class="post-list-filter__head">
+
+		<h3>
+			<?php
+			$total_results = $wp_query->found_posts;
+			$paged = ( get_query_var( 'paged' ) ) ? get_query_var( 'paged' ) : 1;
+			$posts_per_page = get_query_var( 'posts_per_page' );
+			$first_result = ( $posts_per_page * $paged ) - $posts_per_page + 1;
+			$last_result = min( $total_results, $wp_query->post_count * $paged );
+			if ( $total_results === 1 ) {
+				printf( esc_html__( 'Showing 1 of 1 result', 'shiro' ) );
+			} elseif ( $total_results > 0 ) {
+				printf(
+					/* translators: 1. first result, 2. last result, 3. total results */
+					esc_html__( 'Showing %1$s - %2$s of %3$s results', 'shiro' ),
+					esc_html( $first_result ),
+					esc_html( $last_result ),
+					esc_html( $total_results )
+				);
+			}
+			?>
+		</h3>
+
+		<button class="post-list-filter__toggle">
+			<?php
+			$filter_count = 0;
+
+			if ( isset( $_GET['post_list_filters_nonce'] ) && wp_verify_nonce( sanitize_text_field( $_GET['post_list_filters_nonce'] ), 'post_list_filters' ) ) {
+				$search_term = get_search_query();
+				if ( ! empty( $search_term ) ) {
+					$filter_count++;
+				}
+
+				if ( ! empty( sanitize_text_field( $_GET['date_from'] ) ) && ! empty( sanitize_text_field( $_GET['date_to'] ) ) ) {
+					$filter_count++;
+				}
+
+				if ( ! empty( $_GET['categories'] ) ) {
+					$filter_count += count( $_GET['categories'] );
+				}
+
+				printf( esc_html__( 'Show filters', 'shiro' ) );
+
+				if ( $filter_count > 0 ) {
+					echo ' <i>';
+					/* translators: 1. how many filters were applied */
+					printf( esc_html__( '(%s applied)', 'shiro' ), esc_html( $filter_count ) );
+					echo '</i>';
+				}
+			} else {
+				printf( esc_html__( 'Show filters', 'shiro' ) );
+			}
+			?>
+		</button>
+
+	</div>
+
+	<form method="get" class="post-list-filter__form">
+
+		<?php wp_nonce_field( 'post_list_filters', 'post_list_filters_nonce' ); ?>
+
+		<div class="post-list-filter__container mw-980">
+
+			<div class="filter-by-text">
+				<h5>
+					<?php printf( esc_html__( 'Filter by text', 'shiro' ) ); ?>
+					<?php if ( ! empty( get_search_query() ) ) : ?>
+						&nbsp;<i>(<?php printf( esc_html__( 'applied', 'shiro' ) ); ?>)</i>
+					<?php endif; ?>
+				</h5>
+				<div class="search-text-input-button">
+					<input type="text" name="s" value="<?php echo get_search_query(); ?>">
+				</div>
+			</div>
+
+			<div class="filter-by-date">
+				<h5>
+					<?php printf( esc_html__( 'Filter by date', 'shiro' ) ); ?>
+					<?php if ( ! empty( $_GET['date_from'] ) && ! empty( $_GET['date_to'] ) ) : ?>
+						&nbsp;<i>(<?php printf( esc_html__( 'applied', 'shiro' ) ); ?>)</i>
+					<?php endif; ?>
+				</h5>
+				<div class="filter-date-inputs-container">
+					<input type="date" name="date_from" placeholder="From" value="<?php echo isset( $_GET['date_from'] ) ? esc_attr( sanitize_text_field( $_GET['date_from'] ) ) : ''; ?>">
```

Comment:
Yeah, unless we register "date_from" as a query_var, `get_query_var()` won't work here. I'm not aware of a built-in function for retrieving arbitrary GET parameters.

My personal preference is to read all of the query args you'll need into variables at the top of the template part, so that it's obvious what global state it depends on. So at the top of the file you might have something like

```
$date_from = $_GET['date_from'] ?? '';
$date_to = $_GET['date_to] ?? '';
$categories = $_GET['categories'] ?? [];
```

...and then escape them on output.

(This is personal preference though, and I don't think there's anything wrong with the way you've done it or abstracting to the utility function like @kadamwhite suggested.) 

One thing to note though is to be aware of the difference between sanitization and escaping. Use sanitization if you're saving a value to the database, and escaping if you're outputting in HTML. I don't think this `sanitize_text_field` call is necessary here. What might be necessary is `wp_unslash()`, though, to strip slashes that WP adds to all of the REQUEST fields.

Reply:
Thanks @goldenapples, that leaded me to few simplifications more, and I absorbed your pattern of parsing the query vars on the template part header :+1: